### PR TITLE
[Tests-Only] run tests using middleware in CI

### PR DESCRIPTION
**Pull request by dpakach**
## Description

run the tests in CI using the [owncloud-test-middleware](https://github.com/owncloud/owncloud-test-middleware)
We have all the API steps moved to the middleware now so we can now use test middleware to run the tests on web. Currently the code for those steps are in 2 different place which is not really good for maintainability.

## Related Issue

- Related https://github.com/owncloud/web/issues/5132

## How Has This Been Tested?

:robot:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>oc10-integration-notifications</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15596/

<p>


### webUISharingNotifications-shareWithUsers-feature-21.png
  

### webUISharingNotifications-shareWithUsers-feature-32.png
  

### webUISharingNotifications-shareWithUsers-feature-40.png
  

### webUISharingNotifications-shareWithUsers-feature-53.png
  

### webUISharingNotificationsToRoot-shareWithUsers-feature-19.png
  

### webUISharingNotificationsToRoot-shareWithUsers-feature-31.png
  

### webUISharingNotificationsToRoot-shareWithUsers-feature-40.png
  

### webUISharingNotificationsToRoot-shareWithUsers-feature-53.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFilesList</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15596/

<p>


### webUIFilesList-fileList-feature-15.png
  

### webUIFilesList-fileList-feature-22.png
  

### webUIFilesList-fileList-feature-28.png
  

### webUIFilesList-fileList-feature-33.png
  

### webUIFilesList-fileList-feature-41.png
  

### webUIFilesList-fileList-feature-46.png
  

### webUIFilesList-fileList-feature-51.png
  

### webUIFilesList-fileList-feature-60.png
  

### webUIFilesList-fileList-feature-73.png
  

### webUIFilesList-sort-feature-31.png
  

### webUIFilesList-sort-feature-51.png
  

### webUIFilesList-sort-feature-72.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>IntegrationApp2</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15596/

<p>


### webUIMoveFilesFolders-moveFiles-feature-20.png
  

### webUIMoveFilesFolders-moveFiles-feature-51.png
  

### webUIMoveFilesFolders-moveFolders-feature-20.png
  

### webUIMoveFilesFolders-moveFolders-feature-41.png
  

### webUISharingAutocompletion-shareAutocompletion-feature-32.png
  

### webUISharingAutocompletion-shareAutocompletion-feature-44.png
  

### webUISharingInternalUsers-shareWithUsers-feature-57.png
  

### webUISharingInternalUsers-shareWithUsers-feature-58.png
  

### webUISharingInternalUsers-shareWithUsers-feature-59.png
  

### webUISharingInternalUsersToRoot-shareWithUsers-feature-52.png
  

### webUISharingInternalUsersToRoot-shareWithUsers-feature-53.png
  

### webUISharingInternalUsersToRoot-shareWithUsers-feature-54.png
  

### webUISharingPublicBasic-publicLinkCreate-feature-11.png
  

### webUISharingPublicBasic-publicLinkCreate-feature-28.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles-feature-33.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles-feature-34.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles-feature-35.png
  

### webUITrashbinDelete-trashbinDelete-feature-29.png
  

### webUITrashbinFilesFolders-trashbinFilesFolders-feature-23.png
  

### webUITrashbinRestore-trashbinRestore-feature-14.png
  

### webUITrashbinRestore-trashbinRestore-feature-46.png
  

### webUIUpload-upload-feature-15.png
  

### webUIUpload-upload-feature-22.png
  

### webUIUpload-upload-feature-34.png
  

### webUIUpload-upload-feature-48.png
  

### webUIUpload-upload-feature-65.png
  

### webUIUpload-upload-feature-89.png
  

### webUIUpload-upload-feature-98.png
  

### webUIUpload-uploadEdgecases-feature-48.png
  

### webUIUpload-uploadEdgecases-feature-49.png
  

### webUIUpload-uploadEdgecases-feature-50.png
  

### webUIUpload-uploadFileGreaterThanQuotaSize-feature-12.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>IntegrationApp1</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15596/

<p>


### webUICreateFilesFolders-createFolderEdgeCases-feature-60.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-61.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-62.png
  

### webUICreateFilesFolders-createFolders-feature-12.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-12.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-59.png
  

### webUIFavorites-favoritesFile-feature-14.png
  

### webUIFavorites-unfavoriteFile-feature-13.png
  

### webUIFavorites-unfavoriteFile-feature-55.png
  

### webUIFilesCopy-copy-feature-10.png
  

### webUIFilesCopy-copy-feature-36.png
  

### webUIFilesSearch-search-feature-27.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-17.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-9.png
  

### webUIRenameFiles-renameFiles-feature-22.png
  

### webUIRenameFiles-renameFiles-feature-23.png
  

### webUIRenameFiles-renameFiles-feature-24.png
  

### webUIRenameFiles-renameFiles-feature-25.png
  

### webUIRenameFiles-renameFiles-feature-26.png
  

### webUIRenameFiles-renameFiles-feature-43.png
  

### webUIRestrictSharing-disableSharing-feature-16.png
  

### webUIRestrictSharing-restrictReSharing-feature-23.png
  

### webUIRestrictSharing-restrictReSharing-feature-42.png
  

### webUIRestrictSharing-restrictSharing-feature-31.png
  

### webUIRestrictSharing-restrictSharing-feature-40.png
  

### webUIRestrictSharing-restrictSharing-feature-56.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-50.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-72.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>oc10-integration-notifications</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15608/

<p>


### webUISharingNotifications-shareWithUsers-feature-21.png
  

### webUISharingNotifications-shareWithUsers-feature-32.png
  

### webUISharingNotifications-shareWithUsers-feature-40.png
  

### webUISharingNotifications-shareWithUsers-feature-53.png
  

### webUISharingNotificationsToRoot-shareWithUsers-feature-19.png
  

### webUISharingNotificationsToRoot-shareWithUsers-feature-31.png
  

### webUISharingNotificationsToRoot-shareWithUsers-feature-40.png
  

### webUISharingNotificationsToRoot-shareWithUsers-feature-53.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFilesList</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15608/

<p>


### webUIFilesList-fileList-feature-15.png
  

### webUIFilesList-fileList-feature-22.png
  

### webUIFilesList-fileList-feature-28.png
  

### webUIFilesList-fileList-feature-33.png
  

### webUIFilesList-fileList-feature-41.png
  

### webUIFilesList-fileList-feature-46.png
  

### webUIFilesList-fileList-feature-51.png
  

### webUIFilesList-fileList-feature-60.png
  

### webUIFilesList-fileList-feature-73.png
  

### webUIFilesList-sort-feature-31.png
  

### webUIFilesList-sort-feature-51.png
  

### webUIFilesList-sort-feature-72.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>IntegrationApp1</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15608/

<p>


### webUICreateFilesFolders-createFolderEdgeCases-feature-60.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-61.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-62.png
  

### webUICreateFilesFolders-createFolders-feature-12.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-12.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-59.png
  

### webUIFavorites-favoritesFile-feature-14.png
  

### webUIFavorites-unfavoriteFile-feature-13.png
  

### webUIFavorites-unfavoriteFile-feature-55.png
  

### webUIFilesCopy-copy-feature-10.png
  

### webUIFilesCopy-copy-feature-36.png
  

### webUIFilesSearch-search-feature-27.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-17.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-9.png
  

### webUIRenameFiles-renameFiles-feature-22.png
  

### webUIRenameFiles-renameFiles-feature-23.png
  

### webUIRenameFiles-renameFiles-feature-24.png
  

### webUIRenameFiles-renameFiles-feature-25.png
  

### webUIRenameFiles-renameFiles-feature-26.png
  

### webUIRenameFiles-renameFiles-feature-43.png
  

### webUIRestrictSharing-disableSharing-feature-16.png
  

### webUIRestrictSharing-restrictReSharing-feature-23.png
  

### webUIRestrictSharing-restrictReSharing-feature-42.png
  

### webUIRestrictSharing-restrictSharing-feature-31.png
  

### webUIRestrictSharing-restrictSharing-feature-40.png
  

### webUIRestrictSharing-restrictSharing-feature-56.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-50.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-72.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>IntegrationApp2</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15608/

<p>


### webUIMoveFilesFolders-moveFiles-feature-20.png
  

### webUIMoveFilesFolders-moveFiles-feature-51.png
  

### webUIMoveFilesFolders-moveFolders-feature-20.png
  

### webUIMoveFilesFolders-moveFolders-feature-41.png
  

### webUISharingAutocompletion-shareAutocompletion-feature-32.png
  

### webUISharingAutocompletion-shareAutocompletion-feature-44.png
  

### webUISharingInternalUsers-shareWithUsers-feature-57.png
  

### webUISharingInternalUsers-shareWithUsers-feature-58.png
  

### webUISharingInternalUsers-shareWithUsers-feature-59.png
  

### webUISharingInternalUsersToRoot-shareWithUsers-feature-52.png
  

### webUISharingInternalUsersToRoot-shareWithUsers-feature-53.png
  

### webUISharingInternalUsersToRoot-shareWithUsers-feature-54.png
  

### webUISharingPublicBasic-publicLinkCreate-feature-11.png
  

### webUISharingPublicBasic-publicLinkCreate-feature-28.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles-feature-33.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles-feature-34.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles-feature-35.png
  

### webUITrashbinDelete-trashbinDelete-feature-29.png
  

### webUITrashbinFilesFolders-trashbinFilesFolders-feature-23.png
  

### webUITrashbinRestore-trashbinRestore-feature-14.png
  

### webUITrashbinRestore-trashbinRestore-feature-46.png
  

### webUIUpload-upload-feature-15.png
  

### webUIUpload-upload-feature-22.png
  

### webUIUpload-upload-feature-34.png
  

### webUIUpload-upload-feature-48.png
  

### webUIUpload-upload-feature-65.png
  

### webUIUpload-upload-feature-89.png
  

### webUIUpload-upload-feature-98.png
  

### webUIUpload-uploadEdgecases-feature-48.png
  

### webUIUpload-uploadEdgecases-feature-49.png
  

### webUIUpload-uploadEdgecases-feature-50.png
  

### webUIUpload-uploadFileGreaterThanQuotaSize-feature-12.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>oc10-integration-notifications</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15609/

<p>


### webUISharingNotifications-shareWithUsers-feature-21.png
  

### webUISharingNotifications-shareWithUsers-feature-32.png
  

### webUISharingNotifications-shareWithUsers-feature-40.png
  

### webUISharingNotifications-shareWithUsers-feature-53.png
  

### webUISharingNotificationsToRoot-shareWithUsers-feature-19.png
  

### webUISharingNotificationsToRoot-shareWithUsers-feature-31.png
  

### webUISharingNotificationsToRoot-shareWithUsers-feature-40.png
  

### webUISharingNotificationsToRoot-shareWithUsers-feature-53.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFilesList</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15609/

<p>


### webUIFilesList-fileList-feature-15.png
  

### webUIFilesList-fileList-feature-22.png
  

### webUIFilesList-fileList-feature-28.png
  

### webUIFilesList-fileList-feature-33.png
  

### webUIFilesList-fileList-feature-41.png
  

### webUIFilesList-fileList-feature-46.png
  

### webUIFilesList-fileList-feature-51.png
  

### webUIFilesList-fileList-feature-60.png
  

### webUIFilesList-fileList-feature-73.png
  

### webUIFilesList-sort-feature-31.png
  

### webUIFilesList-sort-feature-51.png
  

### webUIFilesList-sort-feature-72.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>IntegrationApp2</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15609/

<p>


### webUIMoveFilesFolders-moveFiles-feature-20.png
  

### webUIMoveFilesFolders-moveFiles-feature-51.png
  

### webUIMoveFilesFolders-moveFolders-feature-20.png
  

### webUIMoveFilesFolders-moveFolders-feature-41.png
  

### webUISharingAutocompletion-shareAutocompletion-feature-32.png
  

### webUISharingAutocompletion-shareAutocompletion-feature-44.png
  

### webUISharingInternalUsers-shareWithUsers-feature-57.png
  

### webUISharingInternalUsers-shareWithUsers-feature-58.png
  

### webUISharingInternalUsers-shareWithUsers-feature-59.png
  

### webUISharingInternalUsersToRoot-shareWithUsers-feature-52.png
  

### webUISharingInternalUsersToRoot-shareWithUsers-feature-53.png
  

### webUISharingInternalUsersToRoot-shareWithUsers-feature-54.png
  

### webUISharingPublicBasic-publicLinkCreate-feature-11.png
  

### webUISharingPublicBasic-publicLinkCreate-feature-28.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles-feature-33.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles-feature-34.png
  

### webUISharingPublicDifferentRoles-shareByPublicLinkDifferentRoles-feature-35.png
  

### webUITrashbinDelete-trashbinDelete-feature-29.png
  

### webUITrashbinFilesFolders-trashbinFilesFolders-feature-23.png
  

### webUITrashbinRestore-trashbinRestore-feature-14.png
  

### webUITrashbinRestore-trashbinRestore-feature-46.png
  

### webUIUpload-upload-feature-15.png
  

### webUIUpload-upload-feature-22.png
  

### webUIUpload-upload-feature-34.png
  

### webUIUpload-upload-feature-48.png
  

### webUIUpload-upload-feature-65.png
  

### webUIUpload-upload-feature-89.png
  

### webUIUpload-upload-feature-98.png
  

### webUIUpload-uploadEdgecases-feature-48.png
  

### webUIUpload-uploadEdgecases-feature-49.png
  

### webUIUpload-uploadEdgecases-feature-50.png
  

### webUIUpload-uploadFileGreaterThanQuotaSize-feature-12.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>IntegrationApp1</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15609/

<p>


### webUICreateFilesFolders-createFolderEdgeCases-feature-60.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-61.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-62.png
  

### webUICreateFilesFolders-createFolders-feature-12.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-12.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-59.png
  

### webUIFavorites-favoritesFile-feature-14.png
  

### webUIFavorites-unfavoriteFile-feature-13.png
  

### webUIFavorites-unfavoriteFile-feature-55.png
  

### webUIFilesCopy-copy-feature-10.png
  

### webUIFilesCopy-copy-feature-36.png
  

### webUIFilesSearch-search-feature-27.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-17.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-9.png
  

### webUIRenameFiles-renameFiles-feature-22.png
  

### webUIRenameFiles-renameFiles-feature-23.png
  

### webUIRenameFiles-renameFiles-feature-24.png
  

### webUIRenameFiles-renameFiles-feature-25.png
  

### webUIRenameFiles-renameFiles-feature-26.png
  

### webUIRenameFiles-renameFiles-feature-43.png
  

### webUIRestrictSharing-disableSharing-feature-16.png
  

### webUIRestrictSharing-restrictReSharing-feature-23.png
  

### webUIRestrictSharing-restrictReSharing-feature-42.png
  

### webUIRestrictSharing-restrictSharing-feature-31.png
  

### webUIRestrictSharing-restrictSharing-feature-40.png
  

### webUIRestrictSharing-restrictSharing-feature-56.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-50.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-72.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFilesList</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15612/

<p>


### webUIFilesList-fileList-feature-15.png
  

### webUIFilesList-fileList-feature-22.png
  

### webUIFilesList-fileList-feature-28.png
  

### webUIFilesList-fileList-feature-33.png
  

### webUIFilesList-fileList-feature-41.png
  

### webUIFilesList-fileList-feature-46.png
  

### webUIFilesList-fileList-feature-51.png
  

### webUIFilesList-fileList-feature-60.png
  

### webUIFilesList-fileList-feature-73.png
  

### webUIFilesList-sort-feature-51.png
  

### webUIFilesList-sort-feature-72.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Move</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15618/

<p>


### webUIMoveFilesFolders-moveFiles-feature-105.png
  

### webUIMoveFilesFolders-moveFiles-feature-115.png
  

### webUIMoveFilesFolders-moveFiles-feature-12.png
  

### webUIMoveFilesFolders-moveFiles-feature-140.png
  

### webUIMoveFilesFolders-moveFiles-feature-20.png
  

### webUIMoveFilesFolders-moveFiles-feature-43.png
  

### webUIMoveFilesFolders-moveFiles-feature-51.png
  

### webUIMoveFilesFolders-moveFiles-feature-79.png
  

### webUIMoveFilesFolders-moveFiles-feature-81.png
  

### webUIMoveFilesFolders-moveFiles-feature-82.png
  

### webUIMoveFilesFolders-moveFiles-feature-85.png
  

### webUIMoveFilesFolders-moveFiles-feature-97.png
  

### webUIMoveFilesFolders-moveFolders-feature-12.png
  

### webUIMoveFilesFolders-moveFolders-feature-20.png
  

### webUIMoveFilesFolders-moveFolders-feature-33.png
  

### webUIMoveFilesFolders-moveFolders-feature-41.png
  

### webUIMoveFilesFolders-moveFolders-feature-66.png
  

### webUIMoveFilesFolders-moveFolders-feature-68.png
  

### webUIMoveFilesFolders-moveFolders-feature-69.png
  

### webUIMoveFilesFolders-moveFolders-feature-72.png
  

### webUIMoveFilesFolders-moveFolders-feature-81.png
  

### webUIMoveFilesFolders-moveFolders-feature-88.png
  


</p></details>

**Comment by kiranparajuli589** [comment]
lgtm 👍🏼

**Review by individual-it — COMMENTED** [review]
Looks good, some comments:

- should we use `in the server` or `on the server` as in the desktop client?  CC @phil-davis 
- documentation about starting selenium needs to be refactored, because now selenium will be started from the docker-compose file
- need to be tested on Mac, maybe by @LukasHirt ?

**Comment by LukasHirt** [comment]
> need to be tested on Mac, maybe by @LukasHirt ?

I could try it tomorrow if that's enough

File: docs/testing.md [hunk 1]
```diff
@@ -62,8 +62,25 @@ In order to run the acceptance tests you need to run ocis using the ocis storage
...
+
+  You can also use docker to start the owncloud test middleware. For that go to `tests/acceptance` and create a `test.env` file with correct values for different test configs. Now run the following command to start the middleware server and selenium server using docker.
+  ```bash
+  # If you are running tests on oc 10
+  docker-compose up -d
+
+  # If  you are running tests on ocis backend
+  docker-compose -f ./docker-compose-ocis.yml up -d
+  ```
+  Or you can run the test middleware directy using docker you can use the following command. Make sure to set the env variables accordingly.
```
Comment (phil-davis) [inline docs/testing.md:76]:
```suggestion
  Or you can run the test middleware directly. You can use the following docker command. Make sure to set the env variables accordingly.
```

**Comment by phil-davis** [comment]
> should we use `in the server` or `on the server` as in the desktop client?

Either word works OK. If we used `on the server` elsewhere, then we should use it here to be consistent.

File: .drone.star [hunk 2]
```diff
@@ -2298,6 +2299,28 @@ def checkStarlark():
         },
     }]
 
+def middlewareService(ocis = False):
+    return [{
+        "name": "middleware",
+        "image": "dpakach/owncloud-test-middleware",
```
[the remark this answers is not in the document]
Comment (dpakach) [reply .drone.star:2305]: Yes, this one is just for the time being so that we can test it

**Comment by LukasHirt** [comment]
@dpakach For MacOS we need to have published ports for the selenium and middleware containers. Currently, if I run the docker-compose I have no way of reaching either of them.

**Comment by dpakach** [comment]
@LukasHirt I have updated the instructions so I think it should work for macOS. Can you try again?

If it still fails can you try once by removing these lines
https://github.com/owncloud/web/pull/5136/files#diff-929940edb8bb2e8a57c6a2637b9529812dc851c6b4f1aff6f1cf8ef367c00f1eR6
https://github.com/owncloud/web/pull/5136/files#diff-929940edb8bb2e8a57c6a2637b9529812dc851c6b4f1aff6f1cf8ef367c00f1eR17

**Comment by LukasHirt** [comment]
@dpakach facing two issues now:

- having the `network_mode: 'host'` in the docker compose fails on Mac because it doesn't support using that mode and port publishing at the same time 🙈 
- removing the network mode and getting the containers to run and then running tests results in `FetchError: request to http://host.docker.internal:3000/init failed, reason: connect ECONNREFUSED 127.0.0.1:3000`. Same error also for `/cleanup`

**Comment by LukasHirt** [comment]
@dpakach Tried now with `MIDDLEWARE_HOST` set to `http://localhost:3000` and still am getting error. This time a different one though.

Screenshot 2021-06-08 at 17 01 06

**Comment by dpakach** [comment]
@LukasHirt can you check the logs for the middleware if there is any hint why it's failing.
you can check that with `docker-compose logs middleware`

**Comment by LukasHirt** [comment]
> can you check the logs for the middleware and see if there is any hint why it's failing.
you can check that with docker-compose logs middleware

@dpakach Nothing to see there. Only two messages about the middleware being started but nothing else.

**Comment by butonic** [comment]
@dpakach @LukasHirt @saw-jan what are the next steps?

**Comment by LukasHirt** [comment]
IMHO if it works on linux it would still be possible to run tests without it on other platforms I would say let's move forward and take care of other platforms in separate PRs.

**Comment by kulmann** [comment]
Is this still relevant / being worked on?

**Comment by pascalwengerter** [comment]
> Is this still relevant / being worked on?

@dpakach any feedback? :)

**Comment by dpakach** [comment]
> > Is this still relevant / being worked on?
> 
> @dpakach any feedback? :)

@pascalwengerter sorry I missed the message. since we have the new docker setup for running the tests, I'm trying to plug the middleware into that setup and with that it will probably work on mac systems as well. I'll try to work this out in coming days. 👍🏼

**Comment by ownclouders** [comment]
Results for <strong>oCISFilesList</strong> https://drone.owncloud.com/owncloud/web/18904/8/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISFilesList</strong> https://drone.owncloud.com/owncloud/web/19180/8/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by dpakach** [comment]
> the concept looks good for me. Do we want to delete all the steps out of web in this PR or in a separate PR?

That will be a big PR so should be done in small parts IMO. But first we need to verify this works in mac.

**Comment by ScharfViktor** [comment]
> > the concept looks good for me. Do we want to delete all the steps out of web in this PR or in a separate PR?
> 
> That will be a big PR so should be done in small parts IMO. But first we need to verify this works in mac.

@dpakach I can check how it works on mac. Please do a rebase

**Comment by dpakach** [comment]
@ScharfViktor thanks a lot. I have rebased it please have a look.

**Comment by ScharfViktor** [comment]
### run tests in ocis
docker with macOC Big Sure 11.4 

`docker-compose up ocis selenium vnc middleware-ocis`
`yarn test:acceptance:ocis features/webUIFilesList/fileList.feature`
test run was successful, the test data was created successfully with the middleware 
log middleware-ocis: 
Screen Shot 2021-11-17 at 3 41 42 PM

### run tests in oc10
docker with macOC Big Sure 11.4 
`docker-compose up oc10 selenium vnc middleware-oc10`
`yarn test:acceptance:oc10 features/webUIFilesList/fileList.feature`
the test data was created successfully with the middleware, but test was failed:
`✖ And user "Alice" has logged in using the webUI # stepDefinitions/loginContext.js:70
       Timed out while waiting for element <button[type=submit]> `
image
log middleware-oc10: 
image

**Comment by dpakach** [comment]
@ScharfViktor thanks a lot for checking this. Before running the tests you should also need to build and bundle web with 
`yarn build:w` before you start the docker setup.
But since it worked on ocis it should also work with oc10.

File: .drone.star (hunk 2 above)
[the remark this answers is not in the document]
Comment (phil-davis) [reply .drone.star:2740]: what do we do about this now?
"copy" into owncloud-ci?

**Review by individual-it — CHANGES_REQUESTED** [review]
the middleware docker should be on owncloud-ci, the rest looks good

File: docker-compose.yml [hunk 3]
```diff
@@ -62,6 +62,38 @@ services:
...
+    hostname: middleware
+    extra_hosts:
+      - host.docker.internal:${DOCKER_HOST:-host-gateway}
+    ports:
+      - "3000:3000"
+    volumes:
+      - /var/tmp/ocis:/var/tmp/ocis
+      - ./tests/acceptance/filesForUpload:/usr/src/app/filesForUpload
+    environment:
+      MIDDLEWARE_PORT: 3000
```
Comment (pascalwengerter) [inline docker-compose.yml:76]: Can we use different ports here so we can start both the ocis-middleware and the oc10-middleware at the same time? :)

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/20516/17/1
The following scenarios passed on retry:
- webUIFilesActionMenu/versions.feature:15

**Review by individual-it — APPROVED** [review]

**Pull request merged**